tests/integration/utils2.go

```diff
@@ -1204,90 +1207,136 @@ func createDoc(
+			docs, err = client.NewDocsFromJSON(bytes, collections[action.CollectionID].Definition())
+		} else {
+			doc, err = client.NewDocFromJSON(bytes, collections[action.CollectionID].Definition())
+			docs = []*client.Document{doc}
+		}
 	}
 	if err != nil {
 		return nil, err
 	}
 
 	txn := getTransaction(s, node, immutable.None[int](), action.ExpectedError)
 
-	ctx := db.SetContextTxn(s.ctx, txn)
-	ctx = db.SetContextIdentity(ctx, action.Identity)
+	ctx := makeContextForDocCreate(db.SetContextTxn(s.ctx, txn), &action)
 
-	return doc, collections[action.CollectionID].Save(ctx, doc)
+	for _, doc := range docs {
+		err = collections[action.CollectionID].Save(ctx, doc)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return docs, nil
+}
+
+func makeContextForDocCreate(ctx context.Context, action *CreateDoc) context.Context {
+	ctx = db.SetContextIdentity(ctx, action.Identity)
+	if action.IsEncrypted {
+		ctx = encryption.SetContextConfig(ctx, encryption.DocEncConfig{IsEncrypted: true})
+	}
+	return ctx
 }
 
 func createDocViaColCreate(
 	s *state,
 	action CreateDoc,
 	node client.DB,
 	collections []client.Collection,
-) (*client.Document, error) {
-	var err error
+) ([]*client.Document, error) {
+	var docs []*client.Document
 	var doc *client.Document
+	var err error
 	if action.DocMap != nil {
 		doc, err = client.NewDocFromMap(action.DocMap, collections[action.CollectionID].Definition())
+		docs = []*client.Document{doc}
 	} else {
-		doc, err = client.NewDocFromJSON([]byte(action.Doc), collections[action.CollectionID].Definition())
+		if client.IsJSONArray([]byte(action.Doc)) {
+			docs, err = client.NewDocsFromJSON([]byte(action.Doc), collections[action.CollectionID].Definition())
+		} else {
+			doc, err = client.NewDocFromJSON([]byte(action.Doc), collections[action.CollectionID].Definition())
+			docs = []*client.Document{doc}
+		}
 	}
 	if err != nil {
 		return nil, err
 	}
 
 	txn := getTransaction(s, node, immutable.None[int](), action.ExpectedError)
 
-	ctx := db.SetContextTxn(s.ctx, txn)
-	ctx = db.SetContextIdentity(ctx, action.Identity)
+	ctx := makeContextForDocCreate(db.SetContextTxn(s.ctx, txn), &action)
+
+	if len(docs) > 1 {
+		err = collections[action.CollectionID].CreateMany(ctx, docs)
+	} else {
+		err = collections[action.CollectionID].Create(ctx, doc)
+	}
 
-	return doc, collections[action.CollectionID].Create(ctx, doc)
+	return docs, err
 }
 
 func createDocViaGQL(
 	s *state,
 	action CreateDoc,
 	node client.DB,
 	collections []client.Collection,
-) (*client.Document, error) {
+) ([]*client.Document, error) {
 	collection := collections[action.CollectionID]
-	var err error
 	var input string
 
+	paramName := "input"
+
+	var err error
 	if action.DocMap != nil {
 		input, err = valueToGQL(action.DocMap)
+	} else if client.IsJSONArray([]byte(action.Doc)) {
+		var docMaps []map[string]any
+		err = json.Unmarshal([]byte(action.Doc), &docMaps)
+		require.NoError(s.t, err)
+		paramName = "inputs"
```

Comment:
todo: This should really be defined in client/request/const.go IMO

Reply:
done